lib/manageiq/api/common/rbac/roles.rb

```diff
@@ -0,0 +1,77 @@
+module ManageIQ
+  module API
+    module Common
+      module RBAC
+        class Roles
+          attr_reader :roles
+
+          def initialize(prefix = nil, scope = 'principal')
+            @roles = {}
+            load(prefix, scope)
+          end
+
+          def find(name)
+            uuid = @roles[name]
+            get(uuid) if uuid
+          end
+
+          def with_each_role
+            @roles.each_value do |uuid|
+              yield get(uuid)
```

Comment:
yikes...we hit the API in a loop?

Reply:
The `@roles` here already filtered by `prefix` and current user.